src/main/java/org/jacp/controller/CreateCompetitionController.java

```diff
@@ -0,0 +1,61 @@
+package org.jacp.controller;
+
+import org.jacp.dto.CompetitionDto;
+import org.jacp.dto.ParticipantDto;
+import org.jacp.dto.QuestionDto;
+import org.jacp.dto.SearchDto;
+import org.jacp.entity.CompetitionEntity;
+import org.jacp.enums.Status;
+import org.jacp.mapper.CompetitionMapper;
+import org.jacp.service.CompetitionService;
+import org.jacp.service.QuestionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.Assert;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+/**
+ * @author saffchen created on 04.12.2023
+ */
+@RestController
+@RequestMapping(value = CreateCompetitionController.URL, produces = MediaType.APPLICATION_JSON_VALUE)
+public class CreateCompetitionController {
+
+    static final String URL = "/api/v1/competition";
+    @Autowired
+    private CompetitionMapper mapper;
+    @Autowired
+    private QuestionService questionService;
+    @Autowired
+    private CompetitionService competitionService;
+
+    @PostMapping("/create")
+    public ResponseEntity<CompetitionEntity> createCompetition(@RequestBody SearchDto searchDto) {
+        List<QuestionDto> questions = questionService.getQuestionsByFilter(searchDto).getBody();
```

Comment:
Can questionService.getQuestionsByFilter(searchDto) immediately return questions list? Without .getBody method call. 
And SearchDto probably can be mapped to FilterDto if you don't use everything from the SearchDto?

Reply:
SearchDtu has more parameters, including user IDs and duration, which are not transferred to CompetitionService, but are needed to create a competition and remain in CompetitionService

Reply:
> Can questionService.getQuestionsByFilter(searchDto) immediately return questions list? Without .getBody method call.

Can not

Reply:
Why cannot? `.getBody` can be executed inside the method, right?

Reply:
```SearchDtu has more parameters, including user IDs and duration, which are not transferred to CompetitionService, but are needed to create a competition and remain in CompetitionService```

Does it make sense to map only required parameters to a new DTO like FilterDto and process it?

Reply:
Fixed